### DISCUSSION TITLE
9th Community Meeting 23/02/2023 - First Meeting of 2023

### DISCUSSION THREAD

**Discussion by VojtechBartoska** [espressif/arduino-esp32#7781] [General]
Hello to all Arduino ESP32 Core Users 🚀 

**Meeting purpose,**
we appreciate all of your contribution and help and we would like to improve your User experience on this Project. 🤩

As a part of this improvement we would like to uncover our internal Development to you, have clear and transparent planning and also listen to your requests, ideas and valuable Feedback.

To fulfill this, we would like to held Bimonthly Community Meeting where we can share main updates and also have an open discussion with you.

---

**💥Ninth Community Meeting will take place on Thursday 23/02/2023 at 4 pm GMT+1 time zone. (1 hour length) We will use Google Meet.**

---

We want to have some idea about how many of you would be interested in joining these Meetings so instead of sharing directly link to Online Meeting itself, it will work by invitation.

**If you want to be added to upcoming Meeting and receive invitation please send an email to [vojtech.bartoska@espressif.com](mailto:vojtech.bartoska@espressif.com).** 

If you already participated on any Community Meeting, you are automatically invited to upcoming Meeting and it's up to you if you join us again or not.

---

### Agenda will be specified below in the comments:  (feel free to add any topic or idea)
1) Development updates & plans (to be specified)
1.1) Improvements of Examples
1.2) Overview of ESP-IDF 5.1 Support
2) Community Topics (to be specified)
2.1) Pull Requests Clean-up
2.2) Issues Clean-up and it's handling
2.3) Licensing
3) Open Discussion

---

See you on the Meeting!

**Comment by VojtechBartoska** [comment]
Hello all, one small change from last year is that we will organize the Community Meetings Bimonthly. The format is going to be the same, using Google Meet. We are looking forward to see you again soon.

**Comment by mrengineer7777** [comment]
Discussion: I would like to see our PRs cleaned up.  What can we do to push them forward or close them?
Personally I'm interested in these:
#7632 Fixes Ethernet events.  I need this in 2 weeks (2023.03.01). [Edit: Merged]
#7765 Fixes a bug in String library.  What is keeping this from merging? [Edit: Merged]
[#6062](https://github.com/espressif/arduino-esp32/pull/6062) Fixing a memory leak seems like a good idea. What would it take to move this forward?

**Reply by mrengineer7777** [reply]
There are also a bunch of issues that we can close out. e.g. wrong repository, answered but user never replied.

**Reply by VojtechBartoska** [reply]
Hi David,

about PRs, we are checking the list from time to time. Some of the mentioned by you have been already resolved by @me-no-dev.

About the issues, you are right and we are aware of this. Recently, issue tracker growth up again and we'll need to clean it a bit and as well find some more systematic approach how to handle them more easily. 

I will write down some point to the issues solving for the Meeting and we can discuss it.

Point to the Agenda added.

**Comment by VojtechBartoska** [comment]
### 2 issues we would like to explain and optionally discuss:
1. [Improvements of Examples](https://github.com/espressif/arduino-esp32/issues/7659)
2. [Overview of ESP-IDF 5.1 Support](https://github.com/espressif/arduino-esp32/issues/7852#top)

**Comment by dalbert2** [comment]
An issue that has come up before but I would like to revisit is commercial-use.  The use of libraries with viral open-source licenses will always be a major limitation for Arduino Core beyond hobby use and will discourage many professional software developers from learning to use it (and therefore contribute to it) since they will rarely be able to use it professionally.  

Ideally, the entire core should use MIT, Apache, or similar unrestricted licenses, but if LGPL must be used, then a solution to allow products containing a mix of open and closed-source code.

Some steps that could be useful would be:
1. Clearly identify the licenses currently used in the core, highlighting which are viral (GPLv3, GPLv2, LGPL, etc.)
2. Document a solution for mixing closed and LGPL (and maybe GPL pre-v3) open-source code: how to separate and provide closed-source objects that can be re-linked with open-source modules.
3. Launch projects (perhaps with a crowd-funding mechanism) to re-implement any portions of the core that are GPL or LGPL, replacing them with a more permissive license.  
4. It might also be worth asking library copyright holders if they would consider less restrictive or commercial licenses.

Some useful references:
- https://forum.arduino.cc/t/lgpl-and-arduino-in-commercial-products/233946/6
- https://forum.arduino.cc/t/software-licensing/525281/2
- https://opensource.stackexchange.com/questions/11365/how-to-use-lgpl-with-embedded-firmwares

**Reply by VojtechBartoska** [reply]
Hello @dalbert2,

thanks for the summary.

Adding this topic to the Meeting Agenda.

**Comment by VojtechBartoska** [comment]
Thanks everyone for very nice Meeting!

I'm sharing the slides here: 
[09_Ninth_Community_meeting-Arduino-February_2023 (1).pdf](https://github.com/espressif/arduino-esp32/files/10815755/09_Ninth_Community_meeting-Arduino-February_2023.1.pdf)
